test.js

```diff
@@ -276,3 +277,54 @@ cmd.title = (message, expected) => `cmd is: ${JSON.stringify(expected)}`;
 test(cmd, ' foo bar', 'foo', 'bar');
 test(cmd, ' baz quz', 'baz', 'quz');
 test(cmd, '');
+
+async function spawnAndKill(t, signal) {
+	const cp = m('sub-process');
+	let pid;
+
+	cp.stdout.setEncoding('utf8');
+	cp.stdout.on('data', chunk => {
+		pid = parseInt(chunk, 10);
+		t.is(typeof pid, 'number');
+
+		setTimeout(() => {
+			process.kill(cp.pid, signal);
+		}, 100);
+	});
+
+	await t.throws(cp);
+
+	// Give everybody some time to breath and kill things
+	const timeout = delay => new Promise(resolve => setTimeout(resolve, delay));
+
+	await timeout(100);
+	t.false(isRunning(cp.pid));
+	t.false(isRunning(pid));
+}
+
+test('cleanup - SIGINT', spawnAndKill, 'SIGINT');
+test('cleanup - SIGKILL', spawnAndKill, 'SIGTERM');
+
+test('cleanup false - SIGINT', async t => {
+	const cp = m('sub-process-false');
+	let pid;
+
+	cp.stdout.setEncoding('utf8');
+	cp.stdout.on('data', chunk => {
+		pid = parseInt(chunk, 10);
+		t.is(typeof pid, 'number');
+
+		setTimeout(() => {
+			process.kill(cp.pid, 'SIGINT');
+		}, 100);
+	});
```

Comment:
This is almost a duplicate of the other test. Can you extract this into a function both could use?